pagefaultgames/pokerogue
# [Ability] (Partially) Implement synchronize ability with old psycho shift interaction

## What are the changes?
Adds (partial) implementation for Synchronize ability as per https://github.com/pagefaultgames/pokerogue/issues/2694. Marking the ability as Partial since this implementation interacts with Psycho Shift the way it was intended before Gen V. Ie:
1. Pyscho shift clears user of status and then transfers to opponent
2. Opponent has synchronize and send the status back to user

There were some race issues with promises that caused buggy behavior with this interaction if we want to implement it the way it was intended post Gen V. That will need more time and code changes to debug, thus doing this for now.

Closes #2694 

## Why am I doing these changes?
More details here: https://github.com/pagefaultgames/pokerogue/issues/2694

## What did change?


- Added new postSetStatusAblAttr ability hook that calls to synchronize status with opposing Pokemon
- Updated Psycho shift logic to clear user status first and then set status on opponent. This shouldn't affect any scenarios other than synchronize.

### Screenshots/Videos
Synchronize working:
![syncInteract](https://github.com/pagefaultgames/pokerogue/assets/28278706/be56e8b3-672a-44d7-adda-4fbff2ddeae5)

Psycho shift interaction (old pre gen V):
![psychoShiftInteract](https://github.com/pagefaultgames/pokerogue/assets/28278706/6f227908-c8bb-4151-9553-abfc94d2aad3)

## How to test the changes?
Added new UTs for the ability and tested manually with overrides on localhost. See videos above. Please set variables in overrides.ts

## Checklist
